### ISSUE TITLE
[2.0-next] Introduce compound parameter type

### ISSUE THREAD

**Issue by patrickpichler** [dynatrace-oss/dynatrace-monitoring-as-code#394]
There should be a new parameter type called `compound`. The main
use case for it is to allow the parameter value to consist of formatted 
string of other parameters. 

It should look something like this:

```yaml
configs:
 - id: overview
   config:
     name: "Overview"
     parameters:
       firstName: "Han"
       sureName: "Solo"
       fullName:
         type: compound
         format: "{{ .firstName }} {{ .sureName }}"
         references:
           - parameter: firstName
           - parameter: sureName
      
```

The value off the `fullName` parameters should then be `Han Solo`.

**Comment by patrickpichler** [comment]
Will be done by @goberchtold

**Comment by patrickpichler** [comment]
Fixed by #395
